### PR TITLE
feat(popup): added a popup component

### LINKED ISSUE #61: [chore] add an alert popup component
Add an alert popup component to inform the user that opening certain links will redirect them to an external app

### PR DESCRIPTION
**Pull request by murosfc**
Closes #61 

The popup component can be customized when called with the following data (url, title, message, width and height).

### CODE REVIEW

**Comment by murosfc** [comment]
@rosostolato updated to Angular cdk

File: src/app/feed/feed-post.component.ts [hunk 1]
```diff
@@ -40,10 +48,37 @@ import { FeedPostAttachmentComponent } from './feed-post-attachment.component';
...
 export class FeedPostComponent {
   readonly post = input.required<FbPost>();
+  readonly popupOpen = signal(false);
+  readonly dialog = inject(Dialog);
+
+  openDialog(): void {
+    this.popupOpen.set(true);
+    const dialogRef = this.dialog.open(PopupComponent, {
+      data: {
+        url: this.post().permalink_url,
```
Comment (rosostolato) [inline src/app/feed/feed-post.component.ts:68]: Remove url

File: src/app/feed/skeleton-feed-post.component.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ import { ChangeDetectionStrategy, Component } from '@angular/core';
           width: 100%;
           height: 300px;
         }
-      }
+      }
```
Comment (rosostolato) [inline src/app/feed/skeleton-feed-post.component.ts:39]: ??

**Review by rosostolato — CHANGES_REQUESTED** [review]
.Asked for small changes

File: package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "@angular/core": "^17.1.0",
     "@angular/fire": "^17.0.1",
     "@angular/forms": "^17.1.0",
+    "@angular/material": "^17.2.0",
```
[the remark this answers is not in the document]
Comment (murosfc) [reply package.json:25]: done!

File: src/app/core/components/popup.component.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    `,
+  ],
+  imports: [ButtonDirective, FormsModule],
+})
+export class PopupComponent {
+  readonly dialogRef = inject(DialogRef);
+  public data: DialogData = inject(DIALOG_DATA);
+
+  close(confirmed?: boolean) {
+    this.dialogRef.close(confirmed ? this.data.url : undefined);
```
[the remark this answers is not in the document]
Comment (murosfc) [reply src/app/core/components/popup.component.ts:70]: forgot to remove after testing

File: src/app/core/components/popup.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        .buttons {
+          display: flex;
+          justify-content: flex-end;
+          padding: 10px;
+          gap: 10px;
+        }
+      }
+    `,
+  ],
+  imports: [ButtonDirective, FormsModule],
```
[the remark this answers is not in the document]
Comment (murosfc) [reply src/app/core/components/popup.component.ts:63]: forgot to remove after changing

File: src/app/feed/feed-post.component.ts [hunk 6]
```diff
@@ -40,10 +48,37 @@ import { FeedPostAttachmentComponent } from './feed-post-attachment.component';
...
+    FormsModule,
+    DialogModule,
+    IconComponent,
+    FeedPostAttachmentComponent,
+  ],
 })
 export class FeedPostComponent {
   readonly post = input.required<FbPost>();
+  readonly popupOpen = signal(false);
+  readonly dialog = inject(Dialog);
```
[the remark this answers is not in the document]
Comment (murosfc) [reply src/app/feed/feed-post.component.ts:62]: my bad!

File: src/app/feed/skeleton-feed-post.component.ts (hunk 2 above)
Comment (murosfc) [reply src/app/feed/skeleton-feed-post.component.ts:39]: ?? lol

**Review by rosostolato — APPROVED** [review]

**Pull request merged**